autopush/router/gcm.py

```diff
@@ -47,6 +47,11 @@ def route_notification(self, notification, uaid_data):
 
     def _route(self, notification, router_data):
         """Blocking GCM call to route the notification"""
+        if not self.gcm:
```

Comment:
This is a dangerous operation, as its running in a thread-pool with no mutex lock. Means that we could end up creating this object multiple times. How come it couldn't be created at start-up?